File: 0058_length_of_last_word/go/solution.go
package length_of_last_word

func lengthOfLastWord(s string) int {
	i := len(s) - 1
	for i >= 0 && s[i] == ' ' {
		i--
	}
	end := i
	for i >= 0 && s[i] != ' ' {
		i--
	}
	return end - i
}

File: 0058_length_of_last_word/kotlin/solution.kt
fun lengthOfLastWord(s: String): Int {
    var i = s.length - 1
    while (i >= 0 && s[i] == ' ') {
        i--
    }

    val end = i
    while (i >= 0 && s[i] != ' ') {
        i--
    }

    return end - i
}
